### PR TITLE
cksum: Add --raw argument

### PR DESCRIPTION
**Pull request by D9nni**
Should close https://github.com/uutils/coreutils/issues/5708

This adds the `--raw` argument. I also added the CkSumError enum to handle error of RawMultipleFiles.

`--raw ` emit a raw binary digest, not hexadecimal

### CODE REVIEW

**Comment by tertsdiepraam** [comment]
I think you forgot to commit the `raw_single_file.expected` file in the `tests/fixtures/cksum` folder. It can't find a file it doesn't know about :)

File: src/uu/cksum/src/cksum.rs [hunk 1]
```diff
@@ -123,7 +151,12 @@ fn cksum<'a, I>(mut options: Options, files: I) -> UResult<()>
 where
     I: Iterator<Item = &'a OsStr>,
 {
-    for filename in files {
+    let files_vec: Vec<_> = files.collect();
```
Comment (cakebaker) [inline src/uu/cksum/src/cksum.rs:154]: I would simply use `files` as variable name, there is no need to add its type to the name.

File: src/uu/cksum/src/cksum.rs [hunk 2]
```diff
@@ -141,6 +174,11 @@ where
         let (sum, sz) = digest_read(&mut options.digest, &mut file, options.output_bits)
             .map_err_context(|| "failed to read input".to_string())?;
 
+        if options.raw {
+            let bytes_str = sum.parse::<u32>().unwrap().to_be_bytes();
```
Comment (cakebaker) [inline src/uu/cksum/src/cksum.rs:178]: Depending on the specified algorithm, this `unwrap()` fails :|  For example:
```
$ cargo run cksum LICENSE --raw -a sm3
thread 'main' panicked at /home/dho/projects/coreutils/src/uu/cksum/src/cksum.rs:178:48:
called `Result::unwrap()` on an `Err` value: ParseIntError { kind: InvalidDigit }
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
It works fine for `crc`, `sysv`, and `bsd`, but not the other algorithms.

Comment (D9nni) [reply src/uu/cksum/src/cksum.rs:178]: Thanks! I will add a test for raw with each algorithm and try to fix that.

**Comment by D9nni** [comment]
Now I made it to work for the other algorithms but it has problems for `crc`, `bsd`, and `sysv`. I think this is because the output for those algorithms isn't generated using encode(). I tried to use parse::<u32> on these cases, but it doesn't seem to work. I will try again, but I would appreciate any suggestions.

**Comment by tertsdiepraam** [comment]
> Now I made it to work for the other algorithms but it has problems for `crc`, `bsd`, and `sysv`. I think this is because the output for those algorithms isn't generated using encode(). I tried to use parse:: on these cases, but it doesn't seem to work. I will try again, but I would appreciate any suggestions.

Is this fixed now? The code looks good to me.

**Comment by D9nni** [comment]
> > Now I made it to work for the other algorithms but it has problems for `crc`, `bsd`, and `sysv`. I think this is because the output for those algorithms isn't generated using encode(). I tried to use parse:: on these cases, but it doesn't seem to work. I will try again, but I would appreciate any suggestions.
> 
> Is this fixed now? The code looks good to me.

Yes, I fixed it and I think it works fine now.

**Comment by sylvestre** [comment]
kudos for `Congrats! The gnu test tests/cksum/cksum-raw is no longer failing!`

**Pull request merged**